Slack.Automation/Promact.Core.Repository/TaskMailRepository/TaskMailRepository.cs

```diff
@@ -359,804 +382,439 @@ public TaskMailRepository(IRepository<TaskMail> taskMail, IStringConstantReposit
             {
-                foreach (var j in jsonResult)
+                foreach (var userRole in userRoleAcList)
                 {
-                    List<TaskMail> taskMails = new List<TaskMail>();
-                    var employeeId = await _userManager.FindByNameAsync(j.UserName);
-                    if (employeeId != null)
-                    {
-                        TaskMailUserAc taskmailUserAc = new TaskMailUserAc
-                        {
-                            UserName = j.Name,
-                            UserId = employeeId.Id,
-                            UserRole = j.Role,
-                            UserEmail = j.UserName
-
-                        };
-                        taskMailUsertAc.Add(taskmailUserAc);
-                    }
+                    TaskMailReportAc taskMailReportAc = new TaskMailReportAc(userRole.UserId,userRole.Role,userRole.Name,userEmail:userRole.UserName);
+                    taskMailReportAcList.Add(taskMailReportAc);
                 }
-            }
-            else if (role.Role == _stringConstant.RoleEmployee)
-            {
-                foreach (var j in jsonResult)
-                {
-                    List<TaskMail> taskMails = new List<TaskMail>();
-                    var employeeId = await _userManager.FindByNameAsync(j.UserName);//_user.FirstOrDefault(x => x.UserName == j.UserName);
-                    if (employeeId != null)
-                    {
-                        TaskMailUserAc taskmailUserAc = new TaskMailUserAc
-                        {
-                            UserName = j.Name,
-                            UserId = employeeId.Id,
-                            UserRole = j.Role,
-                            UserEmail = j.UserName
-                        };
-                        taskMailUsertAc.Add(taskmailUserAc);
-                    }
-                }
-            }
-
-            return taskMailUsertAc;
+           }
+           return taskMailReportAcList;
         }
 
         /// <summary>
         /// Task Mail Details Report Information For the User Role Admin and Employee
         /// </summary>
-        /// <param name="UserId"></param>
-        /// <param name="UserRole"></param>
-        /// <param name="UserName"></param>
-        /// <param name="LoginId"></param>
+        /// <param name="userId"></param>
+        /// <param name="role"></param>
+        /// <param name="userName"></param>
+        /// <param name="loginId"></param>
         /// <returns></returns>
-        public async Task<List<TaskMailUserAc>> GetTaskMailDetailsInformation(string UserId, string UserRole, string UserName, string LoginId)
+        private async Task<List<TaskMailReportAc>> GetTaskMailDetailsInformationAsync(string userId, string role, string userName, string loginId)
```

Comment:
This method also has lots of redundant code which can be extracted to a function.